pinot-core/src/main/java/com/linkedin/pinot/core/query/pruner/AbstractSegmentPruner.java

```diff
@@ -31,7 +32,7 @@
  */
 public abstract class AbstractSegmentPruner implements SegmentPruner {
 
-  public abstract boolean pruneSegment(FilterQueryTree filterQueryTree, Map<String, ColumnMetadata> columnMetadataMap);
+  public abstract boolean pruneSegment(IndexSegment segment, FilterQueryTree filterQueryTree, Map<String, ColumnMetadata> columnMetadataMap);
```

Comment:
I think the lower level functions only need partition info. So instead of passing index segment and columnMetadaMap all over, may be it would be better to extract out that information upfront, and just pass that around.